### DISCUSSION TITLE
scope problem in Lesefeldern

### DISCUSSION THREAD

**Discussion by tolot27** [openjverein/jverein#359] [General]
Kann es sein, dass die Lesefelder oder Beanshell ein Scope Problem hat? Ich habe folgende MWEs:

Skript-Name: test
```java
import java.lang.StringBuilder;
StringBuilder sb = new StringBuilder();
return sb.toString();
```
Speichern, alles okay.

Skript-Name: test2
```java
import java.lang.StringBuffer;
StringBuffer sb = new StringBuffer();
return sb.toString();
```
Speichern führt zu folgendem Skript-Fehler:
`Sourced file: inline evaluation of: ``import java.lang.StringBuffer; StringBuffer sb = new StringBuffer(); return sb.t . . . '' : Typed variable declaration : Typed variable: sb was previously declared with type: class java.lang.StringBuilder`.

Dem Skript `test2` sollte es völlig egal sein, dass sb im Skript `test` mal anders deklariert wurde, es sei denn, es wird eine einzige BSH-Instanz erzeugt, die alle Skripte enthält und am Stück parsed/validiert. Falls nicht, wäre interessant zu wissen, ob es mit BSH 3 behoben ist (siehe #356).

**Comment by lenilsas** [comment]
Es wird anscheinend wirklich nur eine bsh Instanz verwendet. Da du dich anscheinend recht gut mit Java etc. auskennst, kannst du gerne den Code umschreiben und die neue bsh Version testen. Die Datei ist: src/de/jost_net/JVerein/util/LesefeldAuswerter.java
